go-controller/pkg/ovn/controller/admin_network_policy/utils.go

```diff
@@ -0,0 +1,156 @@
+func getAdminNetworkPolicyPGName(name string) (hashedPGName, readablePGName string) {
+	return util.HashForOVN(name), name
+}
+
+// getANPRuleACLDbIDs will return the dbObjectIDs for a given rule's ACLs
+func getANPRuleACLDbIDs(name, gressPrefix, priority, controller string) *libovsdbops.DbObjectIDs {
+	idType := libovsdbops.ACLAdminNetworkPolicy
+	return libovsdbops.NewDbObjectIDs(idType, controller, map[libovsdbops.ExternalIDKey]string{
+		libovsdbops.ObjectNameKey:      name,
+		libovsdbops.PolicyDirectionKey: gressPrefix,
+		// priority is the unique id for address set within given gressPrefix
+		libovsdbops.PriorityKey: priority,
+	})
+}
+
+// getACLActionForANPRule returns the corresponding OVN ACL action for a given ANP rule action
+func getACLActionForANPRule(action anpapi.AdminNetworkPolicyRuleAction) string {
+	var ovnACLAction string
+	switch action {
+	case anpapi.AdminNetworkPolicyRuleActionAllow:
+		ovnACLAction = nbdb.ACLActionAllowRelated
+	case anpapi.AdminNetworkPolicyRuleActionDeny:
+		ovnACLAction = nbdb.ACLActionDrop
+	case anpapi.AdminNetworkPolicyRuleActionPass:
+		ovnACLAction = nbdb.ACLActionPass
+	default:
+		panic(fmt.Sprintf("Failed to build ANP ACL: unknown acl action %s", action))
+	}
+	return ovnACLAction
+}
+
+// getACLDirectionForRule returns the corresponding OVN ACL direction for a given ANP gress rule
+func getACLDirectionForRule(direction string) string {
+	if direction == ANPIngressPrefix {
+		return nbdb.ACLDirectionToLport
+	}
+	return nbdb.ACLDirectionFromLport
+}
+
+// getACLDirectionForRule returns the corresponding OVN ACL options for a given ANP gress rule
+func getACLOptionsForRule(direction string) map[string]string {
+	if direction == ANPEgressPrefix {
+		return map[string]string{
+			"apply-after-lb": "true",
+		}
+	}
+	return make(map[string]string)
+}
+
+// Takes a slice of IPs and returns a slice with string IPs
+func ipsToString(ips []net.IP) []string {
+	s := sets.New[string]()
+	for _, ip := range ips {
+		s.Insert(ip.String())
+	}
+	return s.UnsortedList()
+}
+
+// Takes a slice of strings and returns a slice of net.IPs
+func stringsToIPs(ips []string) []net.IP {
+	s := []net.IP{}
+	for _, ip := range ips {
+		s = append(s, net.ParseIP(ip))
+	}
+	return s
+}
+
+// GetANPPeerAddrSetDbIDs will return the dbObjectIDs for a given rule's address-set
+func GetANPPeerAddrSetDbIDs(name, gressPrefix, priority, controller string) *libovsdbops.DbObjectIDs {
+	idType := libovsdbops.AddressSetAdminNetworkPolicy
+	return libovsdbops.NewDbObjectIDs(idType, controller, map[libovsdbops.ExternalIDKey]string{
+		libovsdbops.ObjectNameKey:      name,
+		libovsdbops.PolicyDirectionKey: gressPrefix,
+		// priority is the unique id for address set within given gressPrefix
+		libovsdbops.PriorityKey: priority,
+	})
+}
+
+// getANPGressPolicyACLName returns the ACL name constructed from a specific rule's fields
+func getANPGressPolicyACLName(anpName, gressPrefix, priority string) string {
```

Comment:
yes I agree, and Nadia also had a good comment here: https://github.com/ovn-org/ovn-kubernetes/pull/3659#discussion_r1257988920 to use BuildACL from ovn/acl package instead of using libovsdbops ACL, I added a TODO here: https://github.com/ovn-org/ovn-kubernetes/pull/3659/commits/b0d0cd137ccda7cc35335d6e30566140f88480b3#diff-64ee751a3bb5ec34f5bf37d2d60bf44ec84150804e64968061d85be31276a4afR285
I plan to do this in the ACL logging PR that I have upcoming...
however the struggle I am having is there is a lot of old history and backwards compatibility issues in the old ACL naming schemes.. let me talk to Nadia and see which functions from ovn/acl.go for names can be reused here and which ones cannot.. basically the problem is its not intuitive this extra layer of abstraction `ovn/acl.go` we have on top of `libovsdb/acl.go`, I will try to spend some time on what commonalities can be leveraged.

Reply:
spoke with Nadia I am going to do a PR to cut out ovn/acl.go into util pkg that both NP/EFW/ANP/BANP can use without any cycle dependency.

Reply:
https://github.com/ovn-org/ovn-kubernetes/pull/3797 - once that merges, I will reuse the utilities in that one here and also move my own pkg/utils into that one perhaps wherever it makes sense...

Reply:
this is done. I am reusing the existing stuff now that the utils can be imported nicely